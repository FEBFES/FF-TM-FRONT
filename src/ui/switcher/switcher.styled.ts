import styled from 'styled-components';

export const SCheckboxInput = styled.div`
  cursor: pointer;
  position: relative;
  transition: 1s;
  width: 32px;
  height: 18px;
  border: 1px solid var(--bg-border);
  border-radius: 10px;
`;

export const SCheckboxCircle = styled.div<{ isActive: boolean }>`
  transition: 0.4s;
  position: absolute;
  width: 12px;
  height: 12px;
  background: ${({ isActive }) => (isActive ? 'var(--bg-primary)' : '#979a9d')};
  border-radius: 100px;
  left: ${({ isActive }) => (isActive ? '16px' : '2px')};
  transform: translateY(-50%);
  top: 50%;
`;
