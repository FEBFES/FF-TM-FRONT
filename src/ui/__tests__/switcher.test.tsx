import React from 'react';
import {Switcher} from '../switcher/switcher';
import renderer from 'react-test-renderer';

describe('<Switcher />', () => {
    it('SNAPSHOTS: UI - Switcher (isActive) component - render', () => {
        const elem = renderer.create(<Switcher isActive={true} onClick={() => {}} />).toJSON();
        expect(elem).toMatchSnapshot();
    });

    it('SNAPSHOTS: UI - Switcher (!isActive) component - render', () => {
        const elem = renderer.create(<Switcher isActive={false} onClick={() => {}} />).toJSON();
        expect(elem).toMatchSnapshot();
    });
})