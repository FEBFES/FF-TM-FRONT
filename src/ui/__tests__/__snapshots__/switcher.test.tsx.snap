// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switcher /> SNAPSHOTS: UI - Switcher (!isActive) component - render 1`] = `
<div
  aria-checked={false}
  aria-disabled={false}
  className="sc-bcPKhP eaxCMA"
  onClick={[Function]}
  role="checkbox"
>
  <div
    className="sc-grXZZQ ezlKrY"
    isActive={false}
  />
</div>
`;

exports[`<Switcher /> SNAPSHOTS: UI - Switcher (isActive) component - render 1`] = `
<div
  aria-checked={true}
  aria-disabled={false}
  className="sc-bcPKhP eaxCMA"
  onClick={[Function]}
  role="checkbox"
>
  <div
    className="sc-grXZZQ kUAa-Dx"
    isActive={true}
  />
</div>
`;
